### PR TITLE
Port code from jest-preset-stylelint

### LINKED ISSUE #1: Port code from jest-preset-stylelint
We should port the test helper code from https://github.com/stylelint/jest-preset-stylelint

### PR DESCRIPTION
**Pull request by ybiquitous**
> Which issue, if any, is this issue related to?

Closes #1

> Is there anything in the PR that needs further explanation?

See https://github.com/stylelint/jest-preset-stylelint

Note: I don't think global features like `getTestRule()` are needed at this time.

### CODE REVIEW

**Comment by ybiquitous** [comment]
Porting is done. Feedback welcome. 🙏🏼

File: lib/index.d.ts [hunk 1]
```diff
@@ -1,16 +1,187 @@
...
+	 */
+	code: string;
+
+	/**
+	 * Description of the test case.
+	 */
+	description?: string;
+
+	/**
+	 * Maps to Jest's `test.only`. Default: `false`.
```
Comment (ybiquitous) [inline lib/index.d.ts:16]: Oops. Forgot to replace "Jest". I'll fix them.

Comment (ybiquitous) [reply lib/index.d.ts]: Done. b05755f

File: CHANGELOG.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+# Changelog
+
+## 0.0.1
```
Comment (jeddy3) [inline CHANGELOG.md:3]:
```suggestion
## 0.1.0
```

**Review by jeddy3 — APPROVED** [review]
Amazing! This is fantastic, thank you.

I've ported stylelint-scales to it:
https://github.com/jeddy3/stylelint-scales/pull/94

It all looks good to me. My only suggestion is to use a minor release number.

I think we can merge, release and then add something to the migration guide, e.g.:

---

If you get a segmentation fault while running the preset on Node.js 18, you can use [jest-light-runner](https://www.npmjs.com/package/jest-light-runner):

```diff js
export default {
  preset: "jest-preset-stylelint",
  setupFiles: ["./jest.setup.js"],
+ runner: "jest-light-runner",
};
```

Alternatively, you can try our new [`stylelint-test-rule-node`](HTTP://npm) package that uses the test runner and assertions built into Node.js:

```diff js
// my-plugin.test.js
+ import { testRule } from "stylelint-test-rule-node";

testRule({
 /* .. */
})
```

```diff json
{
  "scripts": {
+    "test": "node --test"
-    "test": "jest"
  }
}
```

**Comment by ybiquitous** [comment]
Thanks for the review! I'll release the first version `0.1.0` soon. 🚀

**Pull request merged**